controllers/users.js

```diff
@@ -75,6 +78,42 @@ const getUsers = async (req, res) => {
   }
 };
 
+/**
+ * Returns the usernames of inactive/idle users/members
+ *
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ */
+
+const getIdleUsers = async (req, res) => {
+  try {
+    const allUsers = await userQuery.fetchUsers(req.query);
+    const onlyMembers = await members.fetchUsersWithRole(ROLES.MEMBER);
+    const taskParticipants = await tasks.fetchActiveTaskMembers();
+    const idleUsers = allUsers?.filter(({ id }) => !taskParticipants.has(id));
+    const idleUserUserNames = idleUsers?.map((_user) => _user.username);
```

Comment:
If in case we need idle members then we don't need idleUsers and idleUserNames, though here in this code in both cases code will be processed for idleUsers and idleUserNames, So wouldn't it be better if we put these two lines after if block, so in case if we need idle members we just find idle members and return from the there only.

Reply:
Yes, make sense. Thank you for the suggestion. I will make the change.